recirq/qml_lfe/learn_dynamics_C.py

```diff
@@ -0,0 +1,167 @@
+import os
+import cirq
+import cirq_google
+import numpy as np
+import circuit_blocks
+
+from absl import app
+from absl import flags
+from absl import logging
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer("n", None, "System size (total qubits == 2 * n).")
+flags.DEFINE_integer("depth", None, "Circuit depth (block-wise).")
+flags.DEFINE_integer(
+    "n_data",
+    20,
+    "Number of circuits generated for each class (total circuits == 2 * n_data).",
+)
+
+flags.DEFINE_integer(
+    "batch_size",
+    5,
+    "Number of circuits to send over the wire per batch (value does not affect results).",
+)
+
+flags.DEFINE_integer(
+    "n_shots", 2000, "Number of measurements to draw from each individual circuit."
+)
+
+flags.DEFINE_string(
+    "save_dir", "./data", "Path to save experiment data (must already exist)."
+)
+
+flags.DEFINE_bool("use_engine", False, "Whether or not to use quantum engine.")
+
+
+def flatten_circuit(circuit):
+    """Pack operations in circuit to the left as far as possible."""
+    return cirq.Circuit([op for mom in circuit for op in mom])
+
+
+def make_full_circuit(qubit_pairs, tsym, depth):
+    """Generate 1D tsym or scrambling circuit with given depth on qubit_pairs."""
+    inter_gen = None
+    if tsym == 1:
+        # Use tsym gates.
+        inter_gen = circuit_blocks.tsym_block
+    elif tsym == 0:
+        # use scrambling gates.
+        inter_gen = circuit_blocks.scrambling_block
+
+    # Random source for circuits.
+    random_source = np.random.uniform(0, 4, size=(depth * len(qubit_pairs), 2))
+
+    ret_circuit = cirq.Circuit()
+    tmp = circuit_blocks.block_1d_circuit(
+        [qubit[0] for qubit in qubit_pairs], depth, inter_gen, random_source
+    )
+    ret_circuit += flatten_circuit(tmp)
+
+    ret_circuit += cirq.Circuit(cirq.S(qubit[0]) for qubit in qubit_pairs)
+    ret_circuit += cirq.Circuit(cirq.H(qubit[0]) for qubit in qubit_pairs)
+
+    # Merge single qubit gates together and add measurements.
+    cirq.optimizers.merge_single_qubit_gates_into_phxz(ret_circuit)
```

Comment:
I think you can just do `cirq.merge_single_qubit_gates_into_phx`

Reply:
Done.